src/Models/Eloquent/Category.php

```diff
@@ -2,11 +2,23 @@
 
 namespace IndieHD\Velkart\Models\Eloquent;
 
+use Baum\NestedSet\Node;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
 
 class Category extends Model
 {
+    use Node;
+
     public $timestamps = false;
 
-    protected $guarded = ['id'];
+    protected $guarded = ['id', 'parent_id', 'left', 'right', 'depth'];
```

Comment:
So we don't want these fields to be mass assignable ?

Reply:
These are columns for the nested set. Don't think they should be.

Reply:
fair enough